runtime/jvmti/jvmtiGeneral.c

```diff
@@ -177,27 +177,20 @@ jvmtiError JNICALL
 jvmtiGetVersionNumber(jvmtiEnv* env,
 	jint* version_ptr)
 {
-#if JAVA_SPEC_VERSION >= 11
-	J9JavaVM * vm = JAVAVM_FROM_ENV(env);
-#endif /* JAVA_SPEC_VERSION >= 11 */
 	jvmtiError rc = JVMTI_ERROR_NONE;
-	jint rv_version = JVMTI_1_2_3_SPEC_VERSION;
+	jint rv_version =
+#if JAVA_SPEC_VERSION >= 15
+		JVMTI_VERSION_15;
+#elif JAVA_SPEC_VERSION >= 11 /* JAVA_SPEC_VERSION >= 15 */
+		JVMTI_VERSION_11;
+#else /* JAVA_SPEC_VERSION >= 15 */
+		JVMTI_1_2_3_SPEC_VERSION;
+#endif /* JAVA_SPEC_VERSION >= 15 */
```

Comment:
Please don't partially split a declaration like this. Either remove the local variable and make the assignment to `*version_ptr` below version-specific (preferred), or make the declaration (in its entirety) version-specific.

Reply:
Went preferred way.